### PR TITLE
Fix Moffat1D and Moffat2D FWHM

### LINKED ISSUE #8801: Negative FWHM in Moffat1D
I'm making an application to get the best focus for an instrument. I'm using a Moffat1D model to fit the profile and for some reason gamma is negative (which does not matter and I assume it is correct) but when I get the FWHM is negative as well, which as far as I know does not make sense in astronomical images (or any). It seems to me that there is a mistake regarding the formula, see this comment:  https://github.com/astropy/astropy/pull/6027#issuecomment-300380649

from the second-last to the last line 
CodeCogsEqn
and thus  (should be)
CodeCogsEqn_fwhm

and using the notation in [astropy's documentation](http://docs.astropy.org/en/stable/api/astropy.modeling.functional_models.Moffat1D.html#astropy.modeling.functional_models.Moffat1D) the FWHM should be:

CodeCogsEqn_astropy_fwhm

Hopefully someone has  a better insight in this, I will workaround this making `gamma` absolute value always.

I can make a PR if this is really something worth doing.

# reference plot and model

moffat_fitted

The model

```
Model: Moffat1D
Inputs: ('x',)
Outputs: ('y',)
Model set size: 1
Parameters:
      amplitude       x_0          gamma         alpha    
    ------------- ------------ -------------- ------------
    248.392233095 2065.9401953 -90.3411844413 1161.3273732
```

### PR DESCRIPTION
**Pull request by simontorres**
Modified the method `fwhm` to use the absolute value of `gamma` in the calculation of the full width at half maximum of the `Moffat1D` and `Moffat2D` 
fixes #8801

### CODE REVIEW

File: astropy/modeling/tests/test_functional_models.py [hunk 1]
```diff
@@ -117,12 +117,14 @@ def test_Gaussian2D_invalid_inputs():
...
-def test_moffat_fwhm():
+@pytest.mark.parametrize('gamma', (10, -10))
+def test_moffat_fwhm(gamma):
     ans = 34.641016151377542
-    kwargs = {'gamma': 10, 'alpha': 0.5}
+    kwargs = {'gamma': gamma, 'alpha': 0.5}
     m1 = models.Moffat1D(**kwargs)
     m2 = models.Moffat2D(**kwargs)
     assert_allclose([m1.fwhm, m2.fwhm], ans)
+    assert_array_less([0, 0], [m1.fwhm, m2.fwhm])
```
Comment (pllim) [inline astropy/modeling/tests/test_functional_models.py]: Nitpick:
```suggestion
    assert_array_less(0, [m1.fwhm, m2.fwhm])
```

File: CHANGES.rst [hunk 2]
```diff
@@ -43,6 +43,9 @@ astropy.io.votable
 
 astropy.modeling
 ^^^^^^^^^^^^^^^^
+- ``astropy.modeling.functional_models.Moffat1D.fwhm`` and
```
Comment (bsipocz) [inline CHANGES.rst]: no  need for the full name,  saying only ``Moffat1D.fwhm`` and ``Moffat2D.fwhm`` will do.

File: CHANGES.rst [hunk 3]
```diff
@@ -43,6 +43,9 @@ astropy.io.votable
...
+  ``astropy.modeling.functional_models.Moffat2D.fwhm`` will return a positive
+  value when `gamma` is negative. [#8801]
```
Comment (bsipocz) [inline CHANGES.rst]:
```suggestion
  value when ``gamma`` is negative. [#8801]
```

**Comment by pllim** [comment]
Hello. I think the PR is fine but needs some clean-up due to your lucky timing with 3.2 release. Please do the following:

1. Remove duplicate entry in change log.
2. Insert this PR's number (`#8815`) in the entry as well as what you already have.
3. Squash the commits into one. See http://docs.astropy.org/en/latest/development/workflow/development_workflow.html#how-to-squash or feel free to ask for help.

**Comment by simontorres** [comment]
Working on it now. Just one question, in the instructions here http://docs.astropy.org/en/latest/development/workflow/development_workflow.html#add-a-changelog-entry it says you have to put the number of the **issue** it fixes. Should I add the issue and the PR or both?

**Comment by pllim** [comment]
The bot only checks for PR number but you can add both.

**Comment by bsipocz** [comment]
@simontorres - yes, that's a typo in the docs. (the number it refers is even a PR rather than an issue :( )

**Pull request merged**